packages/mdc-menu-surface/index.ts

```diff
@@ -0,0 +1,207 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+import MDCComponent from '@material/base/component';
+import {Corner, CornerBit, cssClasses, MenuDimensions, MenuDistance, MenuPoint, strings} from './constants';
+import {MDCMenuSurfaceFoundation} from './foundation';
+import * as util from './util';
+
+type RegisterFunction = () => void;
+
+class MDCMenuSurface extends MDCComponent<MDCMenuSurfaceFoundation> {
+  static attachTo(root: Element): MDCMenuSurface {
+    return new MDCMenuSurface(root);
+  }
+
+  anchorElement!: Element | null;
+
+  private previousFocus_!: HTMLElement | SVGElement | null;
+  private firstFocusableElement_!: HTMLElement | SVGElement | null;
+  private lastFocusableElement_!: HTMLElement | SVGElement | null;
+
+  private handleKeydown_!: EventListener;
+  private handleBodyClick_!: EventListener;
+  private registerBodyClickListener_!: RegisterFunction;
+  private deregisterBodyClickListener_!: RegisterFunction;
+
+  get open(): boolean {
+    return this.foundation_.isOpen();
+  }
+
+  set open(value: boolean) {
+    if (value) {
+      const focusableElements = this.root_.querySelectorAll<HTMLElement | SVGElement>(strings.FOCUSABLE_ELEMENTS);
+      this.firstFocusableElement_ = focusableElements[0] || null;
+      this.lastFocusableElement_ = focusableElements[focusableElements.length - 1] || null;
+      this.foundation_.open();
+    } else {
+      this.foundation_.close();
+    }
+  }
+
+  set quickOpen(quickOpen: boolean) {
+    this.foundation_.setQuickOpen(quickOpen);
+  }
+
+  initialSyncWithDOM() {
+    const parentEl = this.root_.parentElement;
+    const hasParentAnchor = parentEl && parentEl.classList.contains(cssClasses.ANCHOR);
+
+    this.anchorElement = hasParentAnchor ? parentEl : null;
+    this.previousFocus_ = null;
+    this.firstFocusableElement_ = null;
+    this.lastFocusableElement_ = null;
+
+    if (this.root_.classList.contains(cssClasses.FIXED)) {
+      this.setFixedPosition(true);
+    }
+
+    this.handleKeydown_ = (evt) => this.foundation_.handleKeydown(evt as KeyboardEvent);
```

Comment:
if you use `SpecificEventListener<'keydown'>`, you can remove the cast and just have `(evt) => this.foundation_.handleKeydown(evt)`

This does require us to define `root_!: HTMLElement`

Reply:
Done